packages/delivery-xml-http-request/delivery.js

```diff
@@ -11,20 +11,29 @@ module.exports = (client, win = window) => ({
         if (req.readyState === win.XMLHttpRequest.DONE) {
           const status = req.status
           if (status === 0 || status >= 400) {
-            client._logger.error('Event failed to send…')
+            const err = new Error('Event failed to send')
+            client._logger.error('Event failed to send…', err)
             if (body.length > 10e5) {
               client._logger.warn(`Event oversized (${(body.length / 10e5).toFixed(2)} MB)`)
             }
+            cb(err)
+            return
           }
           cb(null)
         }
       }
+
       req.open('POST', url)
       req.setRequestHeader('Content-Type', 'application/json')
       req.setRequestHeader('Bugsnag-Api-Key', event.apiKey || client._config.apiKey)
       req.setRequestHeader('Bugsnag-Payload-Version', '4')
       req.setRequestHeader('Bugsnag-Sent-At', (new Date()).toISOString())
-      req.send(body)
+      try {
+        req.send(body)
+      } catch (err) {
+        client._logger.error(err)
+        cb(err)
+      }
```

Comment:
I don't think this can throw — it's an asynchronous request, so this won't catch errors that are raised during the request and [MDN says `send` can only throw 2 errors](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/send#exceptions):

- `InvalidStateError` if `send` has already been invoked for the request — this isn't relevant as we can't call `send` multiple times with this code
- `NetworkError` if the resource type to be fetched is a Blob, and the method is not GET — we're not using Blobs and are sending POST requests anyway

So I think this try/catch is unnecessary, unless you found it could throw errors in some other case?

Reply:
I think it only catches the implementation used in the e2e test (same as x-domain), which doesn't really feel right. I would probably prefer to add an onerror handler or event listener... I'll revisit this, particularly as it's one of our main delivery systems for browsers.